external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -221,7 +224,22 @@ private boolean commit() {
     }
 
     private boolean poll() {
-        return !waitingToEmit() && numUncommittedOffsets < kafkaSpoutConfig.getMaxUncommittedOffsets();
+        final int maxUncommittedOffsets = kafkaSpoutConfig.getMaxUncommittedOffsets();
+        final boolean poll = !waitingToEmit() && numUncommittedOffsets < maxUncommittedOffsets;
+
+        if (!poll) {
+            final boolean isLoggable = logTimer.isExpiredResetOnTrue();
```

Comment:
This check is here in order to throttle the number of messages that get printed. Although logs are with debug level, a lot of messages will get printed (one each time nextTuple is called), which will spam the logs, and limit the ability to debug. What this code does is to print one message per second, rather than one per call to nextTuple. By doing this, no info is lost, and things are much cleaner.